src/odemis/gui/cont/tabs/fastem_project_settings_tab.py

```diff
@@ -229,6 +232,7 @@ def _fill_hfw_choices(self):
         )
         hfw_unit = self.tab_data.main.ebeam.horizontalFoV.unit
         hfw_choices_formatted, choices_si_prefix = format_choices(horizontal_fw_choices)
+        hfw_choices_formatted = [(c, f) for c, f in hfw_choices_formatted if c > STAGE_PRECISION]
```

Comment:
It looks to me as you could filter the hfw_choices even earlier: just after computing `horizontal_fw_choices`, filter the list with `c > STAGE_PRECISION`. I think it'd be equivalent, but the code would be simpler (as it's just a list of floats, instead of a tuple). It could also lead to a nicer `choices_si_prefix`.

Reply:
If I filter them first, then for immersion mode I get all choices in `µm` and for non-immersion mode I get all choices in `mm`. For example `50 µm` hfw in immersion is `0.05 mm` hfw in non-immersion. I had like to keep the same ruler scale, so that for a switch between immersion and non-immersion I can also use FindString to find matching combobox values.